edge/services/helloSecretWorld/CreateService.md

```diff
@@ -0,0 +1,262 @@
+
+To make these secret associations, the following JSON snippet is added to the patterns and deployment policies:
+```
+"secretBinding": [
+    {
+        "serviceUrl": "<service-URL>",
+        "serviceOrgid": "<service-org-id>",
+        "serviceArch": "<service-hardware-architecture>,
+        "serviceVersionRange": "x.y.z", 
+        "secrets": [
+            {"<service-secret-name>": "<secrets-manager-secret-name>"}
+        ]
+    }
+]
+```
+
+An example of a service definition, pattern, and deployment policy with secrets are provided in the `horizon` folder of the example service. 
+
+When a deployment pattern or policy that contains a secret association is published to the Exchange, Horizon will check for the existence of all secret names stated in the pattern/policy to be associated to a service's secret name. If a secret name is not found in the secrets manager, Horizon will not be able to publish the pattern or policy to the exchange.
+
+## Using Secrets in the Service Code
+
+### Accessing Secret Details
+
+When an agreement is formed and a service is deployed to a node, a folder called `open-horizon-secrets` is mounted to the service's container. This folder will contain one file for each secret used by a service, and the names of these files with match the name of the secret as specified in the service definition. For example, the secret details for a service secret `password` will be in `open-horizon-secrets/password`. These files will contain the secret details encoded in JSON format as follows:
+```
+{
+    "key": "<secret-key>",
+    "value: "<secret-value>"
+}
+```
+This file is automatically updated if the associated secret in the secrets manager, as specified in the pattern or deployment policy, has its contents updated.
+
+### Updating Secrets
+
+The agent hosts a REST API to notify services when a secret is updated in the secrets manager by an administrator. This API allows the service implementation to query for updated secrets and receive the new details when it is ready to do so. 
+
+The full documentation of this API can be accessed here: https://petstore.swagger.io/?url=https://raw.githubusercontent.com/open-horizon/anax/master/resource/swagger.json. The example service also includes examples of calls to this API for reference.
+
+Each service is provided with an authentication ID and token to access the ESS API. These same credentials are used to authenticate to this API. 
+
+Note: This API is not started on call to `hzn dev service start`, since the agent in that case can only pull secret details from the file in the service's `horizon` subdirectory. If a change is made to that file, it will be reflected in the container as well, but the change will not be communicated to the code if it is only querying the API for changes. A way to get around this would be to continuously read secret details from `open-horizon-secrets`, so there would be need to query the API for changes to secrets.
```

Comment:
The API is available, you just cant update secrets so it will never return anything.